File: corpus/rules/core-security-hardcoded-password/fp/src/good-passwords.py
# False-positive fixture: NOTHING here may fire
# core-security-hardcoded-password.
import os

# Real credential read from the environment — the recommended pattern.
password = os.environ.get("DB_PASSWORD")

# Placeholders and docs values.
db_password = "changeme"
example_password = "your-password-here"
test_password = "xxxxxxxx"
admin_password = "<insert-password>"
template_password = "${DB_PASSWORD}"

# Too short to be a real credential / test stubs.
pw_stub_password = "ab"

# Comment mention only: password = "hunter2secret"

# Identifier does not END with a credential keyword.
password_hash = "5f4dcc3b5aa765d61d8327deb882cf99"
password_field = "password"
password_label = "Enter your password"

File: corpus/rules/core-security-hardcoded-password/fp/src/good-passwords.ts
// False-positive fixture: NOTHING here may fire
// core-security-hardcoded-password.

// Env-var reads — the recommended pattern.
const dbPassword = process.env.DB_PASSWORD;

// UI strings: identifier does not END with a credential keyword.
const passwordPlaceholder = "Enter password";
const inputType = "password";
const passwordSelector = "#password";

// Placeholder values.
const apiKey = "your-api-key";
const clientSecret = "<client-secret>";
const secret = "secret";
const testPassword = "xxxx";

// Template interpolation, not a literal credential.
const connSecret = "${VAULT_SECRET}";

/*
 * Docs example inside a comment: password = "hunter2secret"
 */

export { dbPassword, passwordPlaceholder, inputType, passwordSelector, apiKey, clientSecret, secret, testPassword, connSecret };
